src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -301,12 +331,362 @@ TEST_F(RPCServiceImplTest,
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  EXPECT_CALL(mock_hmi_handler_, SendMessageToHMI(_)).Times(1);
+  message_handler->Handle(am::rpc_service::impl::MessageToHmi(message_to_send));
+}
+
+TEST_F(RPCServiceImplTest, Handle_NoMobileProtocolHandler) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  message_handler->set_protocol_handler(nullptr);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  EXPECT_CALL(mock_protocol_handler_, IsRPCServiceSecure(_)).Times(0);
+  message_handler->Handle(
+      am::rpc_service::impl::MessageToMobile(message_to_send, false));
+}
+
+TEST_F(RPCServiceImplTest, Handle_NoRawMessage) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  EXPECT_CALL(mock_protocol_handler_, IsRPCServiceSecure(_)).Times(0);
+  message_handler->Handle(
+      am::rpc_service::impl::MessageToMobile(message_to_send, false));
+}
+
+TEST_F(RPCServiceImplTest, Handle_NeedsEncryptionAndServiceNotSecure) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  message_to_send->set_protocol_version(
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_1);
+  message_to_send->set_connection_key(kConnectionKey);
+  message_to_send->set_correlation_id(kCorrelationId);
+  message_to_send->set_function_id(kFunctionId);
+  message_to_send->set_message_type(static_cast<am::MessageType>(0));
+  message_to_send->set_json_message(std::string("message"));
+  am::BinaryData binaryData({0, 0, 0});
+  message_to_send->set_binary_data(&binaryData);
+
+  EXPECT_CALL(mock_protocol_handler_, IsRPCServiceSecure(_))
```

Comment:
check arg

Reply:
Fixed in commit 14ea356e21f5d3ab1cd5ee199e11b6d4513f0543